ddl/table.go

```diff
@@ -896,3 +905,46 @@ func checkAddPartitionValue(meta *model.TableInfo, part *model.PartitionInfo) er
 	}
 	return nil
 }
+
+func onRepairTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error) {
+	schemaID := job.SchemaID
+	tblInfo := &model.TableInfo{}
+
+	if err := job.DecodeArgs(tblInfo); err != nil {
+		// Invalid arguments, cancel this job.
+		job.State = model.JobStateCancelled
+		return ver, errors.Trace(err)
+	}
+
+	tblInfo.State = model.StateNone
+
+	// Check the old db and old table exist.
+	_, err := getTableInfo(t, job.TableID, schemaID)
```

Comment:
Could we use `getTableInfoAndCancelFaultJob` directly?

Reply:
Nice catch.